crates/cheatnet/src/execution/gas.rs

```diff
@@ -0,0 +1,29 @@
+use std::collections::HashMap;
+
+use blockifier::{
+    abi::constants, block_context::BlockContext, execution::entry_point::ExecutionResources,
+    fee::fee_utils::calculate_l1_gas_by_vm_usage, transaction::objects::ResourcesMapping,
+};
+use cairo_vm::vm::runners::cairo_runner::ExecutionResources as VmExecutionResources;
+
+#[must_use]
+pub fn recover_gas_from_execution_resources(
+    block_context: &BlockContext,
+    resources: &ExecutionResources,
+) -> f64 {
+    let resource_mapping = vm_execution_resources_to_resource_mapping(&resources.vm_resources);
+    calculate_l1_gas_by_vm_usage(block_context, &resource_mapping)
+        .expect("Calculating gas failed, some resources were not included.")
+}
```

Comment:
I'm not sure how should we validate if it is valid 😅

Reply:
![image](https://github.com/foundry-rs/starknet-foundry/assets/56825108/a842c1bd-7a3e-4a04-8459-3147563b5cac)

(seriously though? no idea)